tls/util.go

```diff
@@ -74,3 +75,29 @@ func parseCertificateRequest(d *schema.ResourceData, pemKey string) (*x509.Certi
 
 	return certReq, nil
 }
+
+func parsePublicKey(d *schema.ResourceData, rsaKey interface{}) error {
```

Comment:
_Minor_ can we change this to `readPublicKey` to imply that it's doing a bit more than just parsing the public key?